.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: SROS2 CI
+on:
+  pull_request:
+  push:
+  schedule:
+    # Run daily
+    - cron:  '0 20 * * *'
+
+jobs:
+  test_latest:
+    runs-on: ubuntu-latest
+    container: osrf/ros2:devel
```

Comment:
What does this actually mean? `runs-on` is the VM used by the action, but then we're saying "please also run the rest of this in the given docker container"?

Reply:
That is my understanding.

`runs-on` is a required element so the workflow file parsing will fail if it's not set
If it's set to anything other than ubuntu, the job will fail because github actions dont support docker on other platforms (hopefully that will change in the future and support windows containers)